src/sentry_core.c

```diff
@@ -191,6 +206,10 @@ sentry_user_consent_t
 sentry_user_consent_get(void)
 {
     sentry__mutex_lock(&g_options_mutex);
+    if (!g_options) {
+        sentry__mutex_unlock(&g_options_mutex);
+        return SENTRY_USER_CONSENT_UNKNOWN;
+    }
     sentry_user_consent_t rv = g_options->user_consent;
     sentry__mutex_unlock(&g_options_mutex);
     return rv;
```

Comment:
This function could probably be simplified by initializing with `rv = SENTRY_USER_CONSENT_UNKNOWN` and then accessing options to overwrite it.